### PR TITLE
Draft: Minor fixes

### PR DESCRIPTION
**Pull request by DreierF**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,13 +41,10 @@ java {
...
 
-repositories {
-    mavenCentral()
-}
-
 dependencies {
-    testImplementation 'org.junit.jupiter:junit-jupiter:5.10.3'
-    testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.10.3'
+    testImplementation(platform('org.junit:junit-bom:5.10.3'))
+    testImplementation('org.junit.jupiter:junit-jupiter')
```
Comment (AnonymFx) [inline build.gradle]: Did you omit a specific version on purpose here (and in the line below)?

File: settings.gradle [hunk 2]
```diff
@@ -6,11 +6,15 @@ pluginManagement {
     }
 }
 
-rootProject.name = 'tia-tutorial'
+plugins {
+    id 'org.gradle.toolchains.foojay-resolver-convention' version '0.8.0'
```
Comment (AnonymFx) [inline settings.gradle]: What's this? Auto-download in case the user does not have the required JVM installed?

File: build.gradle [hunk 3]
```diff
@@ -16,13 +41,10 @@ java {
   }
 }
 
-repositories {
-    mavenCentral()
```
Comment (AnonymFx) [inline build.gradle]: This should probably stay. Also, could you please add `mavenLocal()` here, too. Makes it easier to test unreleased versions of your plugin in the future.
I ran into the following error without it.
```
Execution failed for task ':tiaTests'.
> Error while evaluating property 'agentJvmConfiguration' of task ':tiaTests'.
   > Could not resolve all dependencies for configuration ':teamscaleJaCoCoAgent'.
      > Could not find com.teamscale:teamscale-jacoco-agent:34.0.1-SNAPSHOT.
        Searched in the following locations:
          - https://repo.maven.apache.org/maven2/com/teamscale/teamscale-jacoco-agent/34.0.1-SNAPSHOT/maven-metadata.xml
          - https://repo.maven.apache.org/maven2/com/teamscale/teamscale-jacoco-agent/34.0.1-SNAPSHOT/teamscale-jacoco-agent-34.0.1-SNAPSHOT.pom
        Required by:
            project :

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   You can figure out how project repositories are declared by configuring your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/8.8/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
```

File: build.gradle [hunk 4]
```diff
@@ -1,5 +1,30 @@
 plugins {
     id 'java'
+    id "com.teamscale" version "34.0.1"
```
Comment (AnonymFx) [inline build.gradle]: Where is this addition coming from (the whole teamscale plugin config part)? 🤔 `step7` also has this.

File: build.gradle (hunk 1 above)
Comment (DreierF) [reply build.gradle]: Yes, the versions are given by the platform defined via the bom. This ensures that the versions are aligned i.e. also with the impacted-test-engine dependencies

File: settings.gradle (hunk 2 above)
Comment (DreierF) [reply settings.gradle]: Yes, relying on the installed JVM to be picked up is being removed in Gradle 9.0.

File: build.gradle (hunk 3 above)
Comment (DreierF) [reply build.gradle]: This is defined [here](https://github.com/cqse/tia-tutorial-gradle-junit5/blob/new_step7/settings.gradle#L15) instead. I did try this all locally with the 34.0.1-SNAPSHOT and there it worked 🤔

File: build.gradle (hunk 4 above)
Comment (DreierF) [reply build.gradle]: Super strange. Yeah it is there on step7 and I just updated the minor version. GitHub bug?

File: build.gradle (hunk 3 above)
Comment (DreierF) [reply build.gradle]: Did you maybe add the `repositories` block manually when trying it out?

File: build.gradle (hunk 1 above)
Comment (AnonymFx) [reply build.gradle]: Ah, haven't payed close enough attention to notice `platform` and didn't know the concept of bom yet. Thanks for the explanation! :)

File: build.gradle (hunk 4 above)
Comment (AnonymFx) [reply build.gradle]: Probably 🤷‍♂️

File: build.gradle (hunk 3 above)
Comment (AnonymFx) [reply build.gradle]: Yeah, very possible. My bad then, sorry 🙈

File: settings.gradle (hunk 2 above)
Comment (AnonymFx) [reply settings.gradle]: 👍

**Pull request merged**